### PR TITLE
8917 add qtypes to response feature specs

### PR DESCRIPTION
**Pull request by DevneyHamilton**

### CODE REVIEW

File: spec/features/results/responses/form_rendering_and_submission.rb [hunk 1]
```diff
@@ -28,7 +28,15 @@
               "long_text"
             ]
           }
-        }
+        },
```
Comment (smoyth) [inline spec/features/results/responses/form_rendering_and_submission.rb]: Can you add `_spec` to the end of this file? Thanks!

File: spec/features/results/responses/form_rendering_and_submission.rb [hunk 2]
```diff
@@ -112,8 +128,10 @@
       select2(user.name, from: "response_user_id")
       fill_in_question([0, 0], with: "1")
       drop_in_dropzone(image, 0)
-      select("Animal")
-      select("Dog")
+      within(%(div.node[data-qing-id="#{form.root_group.c[2].id}"])) {
```
[the remark this answers is not in the document]
Comment (smoyth) [reply spec/features/results/responses/form_rendering_and_submission.rb]: Please fix.

File: spec/features/results/responses/form_rendering_and_submission.rb [hunk 3]
```diff
@@ -130,6 +148,18 @@
       fill_in_question([3, 1, 1, 0, 0], with: "7892")
       drop_in_dropzone(image, 2)
       fill_in_question([3, 1, 3], with: "some other text")
+      fill_in_question([4], with: "1.2")
+      fill_in_question([5], with: "3")
+      fill_in_question([6], with: "barcode answer")
+      fill_in_question([7], with: "Dog")
+      # question 8
+      within(%(div.node[data-qing-id="#{form.root_group.c[8].id}"])) {
```
[the remark this answers is not in the document]
Comment (smoyth) [reply spec/features/results/responses/form_rendering_and_submission.rb]: Please fix.

File: spec/features/results/responses/form_rendering_and_submission.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (smoyth) [reply spec/features/results/responses/form_rendering_and_submission.rb]: Actually can you make `fill_in_question` smart enough to accept values like `%w[Animal Dog]` (multilevel) and `%w[Dog Cat]` (multiselect) ? That would make this code a lot smoother.

File: spec/features/results/responses/form_rendering_and_submission.rb [hunk 4]
```diff
@@ -60,7 +68,15 @@
...
             create(:media_image),
             %w[Plant Oak],
-            {repeating: [[2, {repeating: [[3]]}]]}
+            {repeating: [[2, {repeating: [[3]]}]]},
+            1.2,
+            3,
+            "barcode answer",
+            "Dog",
+            %w[Dog Cat],
+            "Mar 12 #{Time.zone.now.year} 18:32:44",
```
Comment (smoyth) [inline spec/features/results/responses/form_rendering_and_submission.rb]: Prefer `#{Time.current.year}` throughout please.

File: spec/support/contexts/response_tree.rb [hunk 5]
```diff
@@ -162,4 +185,13 @@ def expect_read_only_value(path, value)
     el = page.find("[data-path='#{path.join('-')}']")
     expect(el).to have_content(value)
   end
+
+  def temporal_mapping
+    {year: "1i", month: "2i", day: "3i", hour: "4i", minute: "5i", second: "6i"}
+  end
+
+  # Currently works only for paths of depth 1 (children of root)
```
Comment (smoyth) [inline spec/support/contexts/response_tree.rb]: Did you try using `data-path` as above? Would be nice to not have this restriction. The `path_selector` logic is also duplicated in a few places above. Can you pull it out and then use it?

Comment (smoyth) [reply spec/support/contexts/response_tree.rb]: Another thing that may come in handy is the `$=` CSS selector which matches the end of an attribute. So like `input[id$="_datetime_value_5i"]` would match an input with an id attrib with that ending...

**Comment by DevneyHamilton** [comment]
Revised fill_in_question to work with select multiple. Had to fix expect_value for select multiple.

**Comment by DevneyHamilton** [comment]
CI failure was for option set flow, which is a little flaky. Rerunning CI.

File: spec/support/contexts/response_tree.rb [hunk 6]
```diff
@@ -162,4 +177,12 @@ def expect_read_only_value(path, value)
     el = page.find("[data-path='#{path.join('-')}']")
     expect(el).to have_content(value)
   end
+
+  def temporal_mapping
+    {year: "1i", month: "2i", day: "3i", hour: "4i", minute: "5i", second: "6i"}
+  end
+
+  def control_for_temporal(path, type, subfield)
+    find("##{path_selector(path, "#{type}_value_#{temporal_mapping[subfield]}")}")
```
Comment (smoyth) [inline spec/support/contexts/response_tree.rb]: Yes!!

**Pull request merged**